FullStackMVCAPP/FullStackMVCAPP/Controllers/CastleController.cs

```diff
@@ -8,15 +11,18 @@
 namespace FullStackMVCAPP.Controllers
 {
     public class CastleController : Controller
-    {
+    {   
+        private readonly CastleService _CastleService;
+
+        public CastleController()
+        {
+            _CastleService = new CastleService();
+        }
```

Comment:
The controller is calling the `CastleService` directly, this could be altered to use `ICastleService` instead.
```suggestion
        private readonly ICastleService _castleService;

        public CastleController(ICastleService castleService)
        {
            _CastleService = castleService;
        }
```

Reply:
I don't control what is passed to the controller constructor, do I? If not then wouldn't creating this cause issues?